### ISSUE TITLE
Loading Models and Tokenizers From Pretrained - Part 1

### ISSUE THREAD

**Issue by cmarkoya** [cmarkoya/cmarkoya#48]
install: 
torch: this is pytorch. this is the underlying library that is going to execute the matrix computation required for these neural networks that we are downloading.
transformers: this is what we use the grab the model configurations and the model architectures and the tokenizers.
bitsandbytes: this is a library which allows us to quantize our model or basically decrease the size of a model from the memory that the model takes up. 

Step 1: login, make sure we are authorized on the account. 
Step 2: load quantization configuration for model
 -- here we will use bitsandbytes to decrease the floating point precision of our neural network from 32 bits to 4 bits. (trade-off: it also leads to lower precision or lower accuracy in reponses, but not as much as we think, not a 1 to 1 relationship, meaning not 8 times less accurate)
Step 3: load a pre-trained model
import model "AutoModelForCausalLM"
device_map: used to detect if you are using CPU or GPU

Step 4: explore the model configuration
